### PR TITLE
Upgrade to Lucene 9.5.0

### PR DESCRIPTION
**Pull request by javanna**
We have recently upgraded to a Lucene 9.5.0 snapshot. With this commit we upgrade to the final 9.5.0 release.

Main changes are around the float vector field, query and values API.

### CODE REVIEW

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/accesscontrol/FieldExtractor.java [hunk 1]
```diff
@@ -76,8 +75,6 @@ static void extractFields(Query query, Set<String> fields) throws UnsupportedOpe
             fields.add(pointInSetQuery.getField());
         } else if (query instanceof FieldExistsQuery fieldExistsQuery) {
             fields.add(fieldExistsQuery.getField());
-        } else if (query instanceof DocValuesNumbersQuery docValuesNumbersQuery) {
-            fields.add(docValuesNumbersQuery.getField());
```
Comment (javanna) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/accesscontrol/FieldExtractor.java:80]: I am not entirely sure what consequences this has. I do not see a way to retrieve the field from the corresponding query that is generated now in place of the former DocValuesNumbersQuery.

**Review by benwtrent — APPROVED** [review]
I think the vector changes look good. But pinging @jdconrad to double check as many of these code paths were originally implemented by him.

**Comment by benwtrent** [comment]
From what I can tell, the test failures are simply because the `assertEquals` are still wrapping in a `BytesRef` and it should be changed to an `assertArrayEquals` with `byte[]`

**Comment by javanna** [comment]
thanks @benwtrent !

**Pull request merged**